filament/backend/include/backend/platforms/VulkanPlatform.h

```diff
@@ -30,6 +30,13 @@ class VulkanPlatform : public Platform {
     // Given a Vulkan instance and native window handle, creates the platform-specific surface.
     virtual void* createVkSurfaceKHR(void* nativeWindow, void* instance, uint64_t flags) noexcept = 0;
 
+    // We use the extent of the surface to determine the size of the swap chain. For platforms
+    // where vkGetPhysicalDeviceSurfaceCapabilitiesKHR returns undefined surface extent, we use
+    // this fallback as the request size for the swap chain.
+    // This can be a no-op for most platforms.
+    virtual void getSwapChainFallbackExtent(void* nativeWindow, uint32_t* width, uint32_t* height)
+            noexcept {}
```

Comment:
Implement the virtual in a cpp. This is because it is a public API, so the implementation should be "on our side".